website/docs/r/vm_internal_disk.html.markdown

```diff
@@ -0,0 +1,107 @@
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful when connected as sysadmin working across different organisations
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level
+* `vapp_name` - (Required) The vApp this VM internal disk belongs to.
+* `vm_name` - (Required) VM in vApp in which internal disk is created.
+* `allow_vm_reboot` - (Optional) Powers off VM when changing any attribute of an IDE disk or unit/bus number of other disk types, after the change is complete VM is powered back on. Without this setting enabled, such changes on a powered-on VM would fail. Defaults to false.
+* `bus_type` - (Required) The type of disk controller. Possible values: `ide`, `parallel`( LSI Logic Parallel SCSI), `sas`(LSI Logic SAS (SCSI)), `paravirtual`(Paravirtual (SCSI)), `sata`. 
+* `size_in_mb` - (Required) The size of the disk in MB. 
+* `bus_number` - (Required) The number of the SCSI or IDE controller itself.
+* `unit_number` - (Required) The device number on the SCSI or IDE controller of the disk.
+* `iops` - (Optional) Specifies the IOPS for the disk. Default - 0.
+* `storage_profile` - (Optional) Storage profile which overrides the VM default one.
+
+## Attribute reference
+
+* `thin_provisioned` - Specifies whether the disk storage is pre-allocated or allocated on demand.
+
+## Importing
+
+~> **Note:** The current implementation of Terraform import can only import resources into the state. It does not generate
+configuration. [More information.][docs-import]
+
+An existing VM internal disk can be [imported][docs-import] into this resource via supplying its path.
+The path for this resource is made of org-name.vdc-name.vapp-name.vm-name.disk-id
+For example, using this structure, representing a VM internal disk that was **not** created using Terraform:
+
+```hcl
+resource "vcd_vm_internal_disk" "tf-myInternalDisk" {
+  org       = "my-org"
+  vdc       = "my-vdc"
+  vapp_name = "my-vapp"
+  vm_name   = "my-vm" 
+}
+```
+
+You can import such VM internal disk into terraform state using this command
+
+```
+terraform import vcd_vm_internal_disk.tf-myInternalDisk my-org.my-vdc.my-vapp.my-vm.my-disk-id
+```
+
+[docs-import]:https://www.terraform.io/docs/import/
+
+After importing, if you run `terraform plan` you will see the rest of the values and modify the script accordingly for
+further operations.
+
+### Listing VM internal disk IDs
+
+If you want to list IDs there is a special command **`terraform import vcd_vm_internal_disk.imported list@org-name.vcd-name.vapp-name.vm-name`**
+where `org-name` is the organization used, `vdc-name` is vDC name, `vapp-name` is vAPP name and `vm_name` is VM name in that vAPP.
+The output for this command should look similar to the one below:
+
+```shell
+$ terraform import vcd_vm_internal_disk.imported list@org-name.vdc-name.vapp-name.vm-name
```

Comment:
I tried to manage existing root disk (which was previously modified with `override_template_disk`), but got an issue:
1. Defined in config:
```
resource "vcd_vm_internal_disk" "demo_disk_root" {
  vapp_name       = vcd_vapp.demo_vapp.name
  vm_name         = vcd_vapp_vm.demo_vm_wordpress.name
  bus_type        = "parallel"
  size_in_mb      = "11112"
  bus_number      = 0
  unit_number     = 0
  storage_profile = "*"
  allow_vm_reboot = true
}
```
2. Ran list to get ID:
```
$ terraform import vcd_vm_internal_disk.demo_disk_root list@demo-org.demo-vdc.demo-web.demo-vm-wordpress
vcd_vm_internal_disk.demo_disk_root: Importing from ID "list@demo-org.demo-vdc.demo-web.demo-vm-wordpress"...
Retrieving all disks by name
No	ID	BusType		BusNumber	UnitNumber	Size	StoragePofile	Iops	ThinProvisioned
--	--	-------		---------	----------	----	-------------	----	---------------
1	2000	parallel	0		0		11112	*		0	true
2	2016	paravirtual	1		0		512	*		0	true

Error: resource was not imported! resource id must be specified in one of these formats:
'org-name.vdc-name.vapp-name.vm-name.my-internal-disk-id' to import by rule id
'list@org-name.vdc-name.vapp-name.vm-name' to get a list of internal disks with their IDs
```
3. Ran import:
```
$ terraform import vcd_vm_internal_disk.demo_disk_root demo-org.demo-vdc.demo-web.demo-vm-wordpress.2000
vcd_vm_internal_disk.demo_disk_root: Importing from ID "demo-org.demo-vdc.demo-web.demo-vm-wordpress.2000"...
vcd_vm_internal_disk.demo_disk_root: Import complete!
  Imported vcd_vm_internal_disk
vcd_vm_internal_disk.demo_disk_root: Refreshing state... [id=2000]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```
4. Ran plan:
```
  # vcd_vm_internal_disk.demo_disk_root will be updated in-place
  ~ resource "vcd_vm_internal_disk" "demo_disk_root" {
      + allow_vm_reboot  = true
        bus_number       = 0
        bus_type         = "parallel"
        id               = "2000"
        iops             = 0
        size_in_mb       = 11112
        storage_profile  = "*"
        thin_provisioned = true
        unit_number      = 0
        vapp_name        = "demo-web"
        vm_name          = "demo-vm-wordpress"
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```
5. Then apply, but it failed:
```
data.vcd_vapp.demo_vapp: Refreshing state...
vcd_vm_internal_disk.demo_disk_root: Modifying... [id=2000]

Error: error updating VM disks: API Error: 400: [ d17130c6-4de6-460c-89ec-cfa0b0151050 ] The attached disks on VM "demo-vm-wordpress" cannot be modified.

  on xendi.tf line 321, in resource "vcd_vm_internal_disk" "demo_disk_root":
 321: resource "vcd_vm_internal_disk" "demo_disk_root" {
```
What did I do wrong?

Reply:
I have suspicion, but you need show me more information. As according limited info I guess you with import overwrited disk info and it was treated as change and tried to change.

Reply:
OK, so the cause was because I had an independent disk in addition to the internal disk and there was a bug. It has now been fixed in this PR.